daemon/cmd/daemon.go

```diff
@@ -708,16 +708,25 @@ func newDaemon(ctx context.Context, cleaner *daemonCleanup, params *daemonParams
 		return nil, nil, fmt.Errorf("error while opening/creating BPF maps: %w", err)
 	}
 	// Upsert restored CIDRs after the new ipcache has been opened above
-	if len(restoredCIDRidentities) > 0 {
-		d.ipcache.UpsertGeneratedIdentities(restoredCIDRidentities, nil)
+	restoredResource := ipcacheTypes.NewResourceID(ipcacheTypes.ResourceKindDaemon, "", "")
+	if len(d.restoredCIDRs) > 0 {
+		// Asynchronously upsert the prefixes into the new ipcache map.
+		// By the time the datapath gets initialized to consume this
+		// map, ipcache should have completed the async work. If this
```

Comment:
The ipcache won't synchronize until after all k8s caches have synched -- will that be a problem?

Reply:
Hm, that's a good point. AFAIU, a cluster with sufficiently large number of K8s resources will delay CIDR restoration until synced.

Reply:
If the datapath restores with a partial ipcache that's not fully in sync with k8s, that in itself could trigger connectivity or policy enforcement issues for peers. I don't exactly know what the constraints are today but I believe that the datapath should only be replaced after fully syncing with k8s.

Reply:
Yeah, I guess there are security implications with restoring previous state, especially if the IP <-> identity mappings changed. It is more secure for Cilium to not assume that the state of the world was the same as it was when it was last running because it prevents reuse issues.